File: geobus-backend/src/main/java/com/geobus/marrakech/service/AuthService.java
package com.geobus.marrakech.service;

import com.geobus.marrakech.dto.AuthResponse;
import com.geobus.marrakech.dto.LoginRequest;
import com.geobus.marrakech.dto.RegisterRequest;

/**
 * Interface pour les services d'authentification
 */
public interface AuthService {

    /**
     * Enregistre un nouvel utilisateur
     *
     * @param request Données d'enregistrement
     * @return Réponse d'authentification
     */
    AuthResponse register(RegisterRequest request);

    /**
     * Connecte un utilisateur existant
     *
     * @param request Données de connexion
     * @return Réponse d'authentification
     */
    AuthResponse login(LoginRequest request);
}

File: geobusandroid/app/src/main/java/com/geobus/marrakech/network/AuthService.kt
package com.geobus.marrakech.network


import com.geobus.marrakech.model.AuthResponse
import com.geobus.marrakech.model.LoginRequest
import com.geobus.marrakech.model.RegisterRequest
import retrofit2.http.Body
import retrofit2.http.POST

/**
 * Interface pour les requêtes d'authentification
 */
interface AuthService {

    @POST("auth/register")
    suspend fun register(@Body request: RegisterRequest): AuthResponse

    @POST("auth/login")
    suspend fun login(@Body request: LoginRequest): AuthResponse
}